sense-android-library/test/java/nl/sense_os/datastorageengine/CommonSenseProxyTest.java

```diff
@@ -0,0 +1,560 @@
+        String session_id = proxy.loginUser(newUser.get("username"), newUser.get("password"));
+        String name = "test";
+        String displayName = "test";
+        String deviceType = "deviceType";
+        String dataType = "json";
+        String sessionID = session_id;
+        String value = "{\"interval\":0,\"data\":[2.23, 19.45, 20.2]}";
+        String dataStructure = createDataStructure(value);
+
+        // first check the sensor JSONObject has been created,
+        JSONObject sensor = proxy.createSensor(name, displayName, deviceType, dataType, dataStructure, sessionID);
+        assertNotNull("Failed to create a sensor", sensor);
+
+        // then check if the sensorid has been returned from the server
+        String sensorId = sensor.get("sensor_id").toString();
+        assertEquals("The new sensorid returned from the server is empty",false, sensorId.isEmpty());
+
+        JSONArray postData = createDataPoint(sensorId);
+
+        double fromDate = System.currentTimeMillis()/1000;
+        boolean result = proxy.postData(postData,sessionID);
+        assertEquals("Failed to post data to the server",true, result);
+
+        JSONArray getData = proxy.getData(sensorId, fromDate, sessionID);
+        assertEquals("Failed to get data from the server", true, (getData.length()!=0));
+
+    }
+    @Test
+    public void testPostDataAndGetDataWithInvalidParams()throws IOException, JSONException{
+        thrown.expect(IOException.class);
+        thrown.expectMessage("invalid input of date or or sensorID or sessionID");
+        // log in first in order to create sensor
+        String session_id = proxy.loginUser(newUser.get("username"), newUser.get("password"));
+        String name = "test";
+        String displayName = "test";
+        String deviceType = "deviceType";
+        String dataType = "json";
+        String sessionID = session_id;
+        String value = "{\"interval\":0,\"data\":[2.23, 19.45, 20.2]}";
+        String dataStructure = createDataStructure(value);
+
+        // first check the sensor JSONObject has been created,
+        JSONObject sensor = proxy.createSensor(name, displayName, deviceType, dataType, dataStructure, sessionID);
+        assertNotNull("Failed to create a sensor", sensor);
+
+        // then check if the sensorid has been returned from the server
+        String sensorId = sensor.get("sensor_id").toString();
+        assertEquals("The new sensorid returned from the server is empty",false, sensorId.isEmpty());
+
+        JSONArray postData = createDataPoint(sensorId);
+        float fromDate = System.currentTimeMillis()/1000;
+        boolean result = proxy.postData(postData,sessionID);
+        assertEquals("Failed to post data to the server",true, result);
+
+        //get data with null session id
+        JSONArray getData = proxy.getData(sensorId, fromDate, null);
+        assertEquals("Failed to get data from the server", true, (getData.length()!=0));
```

Comment:
In fact, when the session id is null, the exception will be catched, so the test will  not reach the assertion statement. Is this assertion still necessary ? But when length is 0, the message I put is also misleading, I changed that as well,